crates/oxc_linter/src/rules/jest/prefer_called_with.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{
+        collect_possible_jest_call_node, is_type_of_jest_fn_call, parse_expect_jest_fn_call,
+        JestFnKind, PossibleJestNode,
+    },
+};
+
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jest(prefer-called-with): Suggest using `toBeCalledWith()` or `toHaveBeenCalledWith()`.")]
+#[diagnostic(severity(warning), help("Prefer {0:?}With(/* expected args */)"))]
+struct PreferCalledWithDiagnostic(Atom, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferCalledWith;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Suggest using `toBeCalledWith()` or `toHaveBeenCalledWith()`
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    ///
+    /// // valid
+    /// expect(noArgsFunction).toBeCalledWith();
+    /// expect(roughArgsFunction).toBeCalledWith(expect.anything(), expect.any(Date));
+    /// expect(anyArgsFunction).toBeCalledTimes(1);
+    /// expect(uncalledFunction).not.toBeCalled();
+    ///
+    /// // invalid
+    /// expect(someFunction).toBeCalled();
+    /// expect(someFunction).toHaveBeenCalled();
+    /// ```
+    ///
+    PreferCalledWith,
+    style,
+);
+
+impl Rule for PreferCalledWith {
+    fn run_once(&self, ctx: &LintContext<'_>) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferCalledWith {
+    pub fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+
+        if !is_type_of_jest_fn_call(call_expr, possible_jest_node, ctx, &[JestFnKind::Expect]) {
```

Comment:
Feel like we can remove this check, `parse_expect_jest_fn_call` will make sure it.

Reply:
Done~